pennylane/labs/tests/resource_estimation/ops/qubit/test_non_parametric_ops.py

```diff
@@ -72,6 +72,27 @@ def test_resources_from_rep(self):
         assert op.resources(**re.ResourceSWAP.resource_rep().params) == expected
 
 
+class TestS:
+    """Tests for ResourceS"""
+
+    def test_resources(self):
+        """Test that S decomposes into two Ts"""
+        op = re.ResourceS(0)
+        expected = {re.CompressedResourceOp(re.ResourceT, {}): 2}
+        assert op.resources() == expected
+
+    def test_resource_rep(self):
+        """Test that the compressed representation is correct"""
+        expected = re.CompressedResourceOp(re.ResourceS, {})
+        assert re.ResourceS.resource_rep() == expected
+
+    def test_resources_from_rep(self):
+        """Test that the compressed representation yields the correct resources"""
+        op = re.ResourceS(0)
+        expected = {re.CompressedResourceOp(re.ResourceT, {}): 2}
+        assert op.resources(**re.ResourceS.resource_rep().params) == expected
```

Comment:
I like the idea of this test, but we should use the compressed rep for everything to truly test this: 

```suggestion
    def test_resources_from_rep(self):
        """Test that the compressed representation yields the correct resources"""
        op = re.ResourceS(0)
        expected = {re.CompressedResourceOp(re.ResourceT, {}): 2}
        
        op_compressed_rep = op.resource_rep_from_op()
        op_resource_type = op_compressed_rep.op_type
        op_resource_params = op_compressed_rep.params
        assert op_resource_type.resources(op_resource_params) == expected
```

Reply:
Great suggestion! I originally wrote this before we had introduced `resource_rep_from_op`